rust/agents/relayer/src/msg/gelato_submitter.rs

```diff
@@ -1,98 +1,231 @@
-    /// conditions like values in InterchainGasPaymaster.
-    async fn tick(&self) -> Result<()> {
-        // TODO(webbhorn): Pull all available messages out of self.rx
-        // and check if enough gas to process them. If not, put on
-        // wait queue. If there is enough, spawn root task and run
-        // the fwd req op.
+    async fn tick(&mut self) -> Result<()> {
+        // Pull any messages sent by processor over channel.
+        loop {
+            match self.messages.try_recv() {
+                Ok(msg) => {
+                    let op = ForwardRequestOp {
+                        args: self.make_forward_request_args(msg)?,
+                        opts: ForwardRequestOptions::default(),
+                        signer: self.signer.clone(),
+                        http: self.http.clone(),
+                    };
+                    tokio::spawn(async move {
+                        op.run()
+                            .await
+                            .expect("failed unimplemented forward request submit op");
+                    });
+                }
+                Err(TryRecvError::Empty) => {
+                    break;
+                }
+                Err(_) => {
+                    bail!("Disconnected receive channel or fatal err");
+                }
+            }
+        }
+        Ok(())
+    }
 
-        // TODO(webbhorn): Scan pending queue for any newly-eligible
-        // ops and if encountered, spawn them in a root task.
-        // Remove them from pending queue if so.
+    fn make_forward_request_args(&self, msg: SubmitMessageArgs) -> Result<ForwardRequestArgs> {
+        let ivm_base_contract = BaseContract::from(ivm_abi.clone());
+        let call_data = ivm_base_contract.encode(
+            "process",
+            [
+                Token::Address(self.inbox_address),
+                Token::FixedBytes(msg.checkpoint.checkpoint.root.to_fixed_bytes().into()),
+                Token::Uint(msg.checkpoint.checkpoint.index.into()),
+                Token::Array(
+                    msg.checkpoint
+                        .signatures
+                        .iter()
+                        .map(|s| Token::Bytes(s.to_vec()))
+                        .collect(),
+                ),
+                Token::Bytes(msg.committed_message.message.to_vec()),
+                Token::FixedArray(
+                    (0..32)
+                        .map(|i| Token::FixedBytes(msg.proof.path[i].to_vec()))
+                        .collect(),
+                ),
+                Token::Uint(msg.leaf_index.into()),
+            ],
+        )?;
+        Ok(ForwardRequestArgs {
+            chain_id: abacus_domain_to_gelato_chain(self.inbox_domain)?,
+            target: self.ivm_address,
+            data: call_data,
+            fee_token: NATIVE_FEE_TOKEN_ADDRESS,
+            payment_type: PaymentType::AsyncGasTank,
+            max_fee: DEFAULT_MAX_FEE.into(), // Maximum fee that sponsor is willing to pay.
```

Comment:
Is there a reason we are not storing the const as the type that is used here?

Reply:
A fair question -- Gelato wants it as a U256 (a solidity int in the signature hash, and a javascript bigint in the json request), and it seemed less-than-ideal to me to have to represent a literal U256 as a compile-time constant in rust without introducing a third party crate like https://lib.rs/crates/u256-literal. So good old fashioned literal < u32::MAX then .into() seemed simpler to me when writing.

Gonna optimistically hope that's good enough excuse and resolve the conversation, feel free to disagree if it makes a big deal